TransformationSystem/Client/TransformationClient.py

```diff
@@ -346,27 +371,24 @@ def _applyTransformationFilesStateMachine( self, tsFilesAsDict, dictOfProposedLF
     """
     newStatuses = {}
 
-    for lfn in dictOfProposedLFNsStatus.keys():
-      if lfn not in tsFilesAsDict.keys():
-        continue
-      else:
-        newStatus = dictOfProposedLFNsStatus[lfn]
-        # Apply optional corrections
-        if tsFilesAsDict[lfn][0].lower() == 'processed' and dictOfProposedLFNsStatus[lfn].lower() != 'processed':
-          if not force:
-            newStatus = 'Processed'
-        elif tsFilesAsDict[lfn][0].lower() == 'maxreset':
+    for lfn in set( dictOfProposedLFNsStatus ) - set( tsFilesAsDict ):
+      newStatus = dictOfProposedLFNsStatus[lfn]
+      # Apply optional corrections
+      if tsFilesAsDict[lfn][0].lower() == 'processed' and dictOfProposedLFNsStatus[lfn].lower() != 'processed':
+        if not force:
+          newStatus = 'Processed'
+      elif tsFilesAsDict[lfn][0].lower() == 'maxreset':
+        if not force:
+          newStatus = 'MaxReset'
+      elif dictOfProposedLFNsStatus[lfn].lower() == 'unused':
+        errorCount = tsFilesAsDict[lfn][1]
+        # every 10 retries (by default)
+        if errorCount and ( ( errorCount % self.maxResetCounter ) == 0 ):
           if not force:
             newStatus = 'MaxReset'
-        elif dictOfProposedLFNsStatus[lfn].lower() == 'unused':
-          errorCount = tsFilesAsDict[lfn][1]
-          # every 10 retries (by default)
-          if errorCount and ( ( errorCount % self.maxResetCounter ) == 0 ):
-            if not force:
-              newStatus = 'MaxReset'
 
-        if tsFilesAsDict[lfn][0].lower() != newStatus:
-          newStatuses[lfn] = newStatus
+      if tsFilesAsDict[lfn][0].lower() != newStatus:
+        newStatuses[lfn] = newStatus
```

Comment:
These changes to _applyTransformationFileStateMachine make the specific unit test to fail (DIRAC.TransformationSystem.Client.test.test_Client.TransformationClientSuccess.test__applyTransformationFilesStateMachine)

I'll try to understand what happens.

I remind everybody that unit tests should always be run, if present, and should always be written when they are not present.